modules/tensor_mechanics/src/utils/ElasticityTensorR4.C

```diff
@@ -59,13 +63,19 @@ ElasticityTensorR4::elasticJacobianwc(const unsigned int i,
 {
   // d(stress_ij*d(test)/dx_j)/dw_k = d(C_ijmn*eps_mnp*w_p*dtest/dx_j)/dw_k (only nonzero for p=k)
   Real the_sum = 0.0;
+  unsigned int ij1 = i * N * N * N;
   for (unsigned int j = 0; j < N; ++j)
+  {
     for (unsigned int m = 0; m < N; ++m)
       for (unsigned int n = 0; n < N; ++n)
-        the_sum += _vals[i][j][m][n] * PermutationTensor::eps(m, n, k) * grad_test(j);
+        the_sum += _vals[ij1 + m * N + n] * PermutationTensor::eps(m, n, k) * grad_test(j);
+    ij1 += N * N;
```

Comment:
This makes the code quite a bit harder to read. I second Cody's request for actual timings.